openpdf/src/test/java/com/lowagie/text/DocumentTest.java

```diff
@@ -16,19 +16,22 @@ void testThatVersionIsCorrect() {
         // Given
         String versionFromPom = getProjectVersion();
         String versionInCode = Document.getVersion();
-        // maven-bundle-plugin transforms 1.0.5-SNAPSHOT to 1.0.5.SNAPSHOT
-        String versionFromPomNormalized = versionFromPom.replace('-', '.');
+
         // Then
         Assertions.assertTrue(
-            versionInCode.endsWith(versionFromPomNormalized),
+            versionInCode.endsWith(versionFromPom),
             String.format(ERROR_MESSAGE_TEMPLATE, versionInCode, versionFromPom));
     }
 
     private String getProjectVersion() {
-        InputStream is = this.getClass().getClassLoader().getResourceAsStream("version.txt");
+        InputStream is = this.getClass().getClassLoader().getResourceAsStream("com/lowagie/text/version.properties");
         String version = null;
         try (BufferedReader reader = new BufferedReader(new InputStreamReader(is))) {
             version = reader.readLine();
+            int eqPosition = version.indexOf('=');
```

Comment:
Version is retrieved twice (at production and tests), but code is quite different. Consider using `Properties` here too